# MM&A Supermarket Case

## 1. Exploring the Data (EDA)

In [1]:
# importing necessary libraries
import numpy as np
import pandas as pd

In [2]:
# reading in the csv file into a dataframe
file = 'mma_mart.csv'
df = pd.read_csv(file)

In [3]:
# observing the data structure
df.head()

order_id  product_id                                   product_name  \
0         1       49302                               Bulgarian Yogurt   
1         1       11109  Organic 4% Milk Fat Whole Milk Cottage Cheese   
2         1       10246                          Organic Celery Hearts   
3         1       49683                                 Cucumber Kirby   
4         1       43633           Lightly Smoked Sardines in Olive Oil   

   aisle_id                 aisle  department_id    department  
0       120                yogurt             16    dairy eggs  
1       108  other creams cheeses             16    dairy eggs  
2        83      fresh vegetables              4       produce  
3        83      fresh vegetables              4       produce  
4        95   canned meat seafood             15  canned goods

In [4]:
# finding the shape of the data set (row and column)
df.shape

(987259, 7)

The data shows 987259 rows and 7 columns.

In [5]:
# finding amount of unique entries in each column
df.nunique()

order_id         97833
product_id       35070
product_name     35070
aisle_id           134
aisle              134
department_id       21
department          21
dtype: int64

There are ~98000 orders and ~35000 unique products which is across 134 aisles that belong to 21 departments. This shows that selecting 1000-1200 products for the instabasket aisle is selecting only 1000/35000 = ~3% of the total product selection offered at MM&A supermarket.

Next, exploring the aisle and departments to maybe gain some insight on what products should be considered as refrigerant and frozen products.

In [6]:
df['aisle'].unique()

array(['yogurt', 'other creams cheeses', 'fresh vegetables',
       'canned meat seafood', 'fresh fruits', 'packaged cheese', 'eggs',
       'spices seasonings', 'oils vinegars', 'baking ingredients',
       'doughs gelatins bake mixes', 'spreads',
       'packaged vegetables fruits', 'soy lactosefree', 'poultry counter',
       'bread', 'breakfast bakery', 'cold flu allergy',
       'energy granola bars', 'breakfast bars pastries', 'chips pretzels',
       'trail mix snack mix', 'crackers', 'refrigerated',
       'energy sports drinks', 'salad dressing toppings',
       'prepared soups salads', 'milk', 'paper goods',
       'water seltzer sparkling water', 'kosher foods',
       'packaged poultry', 'instant foods', 'packaged produce',
       'cookies cakes', 'candy chocolate', 'body lotions soap',
       'dry pasta', 'laundry', 'air fresheners candles', 'frozen produce',
       'buns rolls', 'canned fruit applesauce', 'juice nectars',
       'granola', 'fresh herbs', 'baby food formul

As can be seen from the aisle names, there are aisles with "frozen" such as `frozen produce` or `frozen meat seafood` in it which all products listed under these aisles would be considered for the 100 frozen product for instabasket. The same can't be said for refridgerant as there a lot of aisles to parse through so it might be better to look at the department to determine what department contains refridgerant products.

In [7]:
df['department'].unique()

array(['dairy eggs', 'produce', 'canned goods', 'pantry', 'meat seafood',
       'bakery', 'personal care', 'snacks', 'breakfast', 'beverages',
       'deli', 'household', 'international', 'dry goods pasta', 'frozen',
       'babies', 'pets', 'alcohol', 'bulk', 'missing', 'other'],
      dtype=object)

From the look at department names, `dairy eggs`, `produce`, `meat seafood`, `deli`, `frozen`, `missing` and `other` are all potential products for fridge and freezer space so it might be better to take a closer look at the products in those department.

In [8]:
dept_list = ['dairy eggs', 'produce', 'meat seafood', 'deli', 'frozen', 'missing', 'other']

In [9]:
# looking at products and aisles of those products for insight as well as how many products each department had
for department in dept_list:
    print(f"Department: {department}")
    print(f"Number of products: {df[df['department']==department]['product_name'].nunique()}\n")
    display(df[df['department']==department][['product_name', 'aisle']])

Department: dairy eggs
Number of products: 2886



product_name  \
0                                        Bulgarian Yogurt   
1           Organic 4% Milk Fat Whole Milk Cottage Cheese   
7                             Organic Whole String Cheese   
8                                      Organic Egg Whites   
17      Total 2% with Strawberry Lowfat Greek Strained...   
...                                                   ...   
987240                        Large Grade AA Organic Eggs   
987241                Reduced Fat Mozarella String Cheese   
987242                   Vanilla Light & Fit Greek Yogurt   
987243               Non-Fat Vanilla Blended Greek Yogurt   
987254            Total 2% All Natural Plain Greek Yogurt   

                       aisle  
0                     yogurt  
1       other creams cheeses  
7            packaged cheese  
8                       eggs  
17                    yogurt  
...                      ...  
987240                  eggs  
987241       packaged cheese  
987242                yogurt  
987243                yogurt  
987254                yogurt  

[164468 rows x 2 columns]

Department: produce
Number of products: 1437



product_name                       aisle
2           Organic Celery Hearts            fresh vegetables
3                  Cucumber Kirby            fresh vegetables
5          Bag of Organic Bananas                fresh fruits
6            Organic Hass Avocado                fresh fruits
9           Michigan Organic Kale            fresh vegetables
...                           ...                         ...
987245                Gala Apples                fresh fruits
987246       Organic Yellow Onion            fresh vegetables
987247       Organic Baby Carrots  packaged vegetables fruits
987249       Organic Baby Spinach  packaged vegetables fruits
987250  Garnet Sweet Potato (Yam)            fresh vegetables

[288775 rows x 2 columns]

Department: meat seafood
Number of products: 692



product_name  \
23      Air Chilled Organic Boneless Skinless Chicken ...   
46               Boneless Skinless Chicken Breast Fillets   
97                            Boneless Beef Sirloin Steak   
176                                      Low Sodium Bacon   
204                  Boneless And Skinless Chicken Breast   
...                                                   ...   
986977                  Organic Air Chilled Whole Chicken   
987067  Natural Hickory Smoked Canadian Bacon Center C...   
987068  All Natural Boneless & Skinless Chicken Breast...   
987175                  Organic Air Chilled Whole Chicken   
987179                      Organic Beef Uncured Hot Dogs   

                         aisle  
23             poultry counter  
46            packaged poultry  
97                meat counter  
176     hot dogs bacon sausage  
204            poultry counter  
...                        ...  
986977         poultry counter  
987067  hot dogs bacon sausage  
987068        packaged poultry  
987175         poultry counter  
987179  hot dogs bacon sausage  

[21588 rows x 2 columns]

Department: deli
Number of products: 1069



product_name                   aisle
40                           Fresh Fruit Salad   prepared soups salads
101                      Mango Pineapple Salsa    fresh dips tapenades
179     Basil, Asiago & Pine Nut Pesto Ravioli          prepared meals
191                             Yuba Tofu Skin  tofu meat alternatives
192                          Organic Firm Tofu  tofu meat alternatives
...                                        ...                     ...
987153                    Spicy Avocado Hummus    fresh dips tapenades
987158                               Guacamole    fresh dips tapenades
987199                           Guacamole Dip    fresh dips tapenades
987204                         Original Hummus    fresh dips tapenades
987253                         Original Hummus    fresh dips tapenades

[32008 rows x 2 columns]

Department: frozen
Number of products: 3127



product_name                    aisle
68                                   Pineapple Chunks           frozen produce
100            Teriyaki & Pineapple Chicken Meatballs             frozen meals
114     All Natural Boneless Skinless Chicken Breasts      frozen meat seafood
118                           Combination Pizza Rolls  frozen appetizers sides
135                    Organic Mini Homestyle Waffles         frozen breakfast
...                                               ...                      ...
987176                 Organic Ice Cream Vanilla Bean            ice cream ice
987180   Dairy Free Coconut Milk Frozen Dessert Minis            ice cream ice
987184                           Organic Mango Chunks           frozen produce
987210                            Bag of Large Lemons      frozen meat seafood
987215                              Vanilla Ice Cream            ice cream ice

[68202 rows x 2 columns]

Department: missing
Number of products: 518



product_name    aisle
654                          Tomato Basil Bisque Soup  missing
1511      Cold Pressed Watermelon & Lemon Juice Blend  missing
1512                           Paleo Blueberry Muffin  missing
2126    Magic Tape Refillable Dispenser 3/4\" x 850\"  missing
4121                           Organic Poblano Pepper  missing
...                                               ...      ...
986067                       Oatneal Cookie Ice Cream  missing
986086    Dairy Free Unsweetened Almond Milk Beverage  missing
986103                Organic Asian Chopped Salad Kit  missing
986275                                      Lemon Bag  missing
986276      Organic Nondairy Strawberry Cashew Yogurt  missing

[2289 rows x 2 columns]

Department: other
Number of products: 303



product_name  aisle
691            Coffee Mate French Vanilla Creamer Packets  other
1077                        SleepGels Nighttime Sleep Aid  other
1926                             Roasted Unsalted Almonds  other
1985                         Camilia, Single Liquid Doses  other
2127    Maximum Strength Original Paste Diaper Rash Oi...  other
...                                                   ...    ...
983992                           Roasted Unsalted Almonds  other
984112                           Boneless Pork Tenderloin  other
984515                              Roasted Almond Butter  other
984534            Light CocoWhip! Coconut Whipped Topping  other
986326                                   93/7 Ground Beef  other

[1097 rows x 2 columns]

Looking at the results, products from `dairy eggs`, `meat seafood` `deli`, and some products from `missing` and `other` (would need to look at what exactly the product is since there was "Oatneal Cookie Ice Cream" in `missing` aisle which is a frozen item and "Boneless Pork Tenderloin" in `other` which is a `meat seafood` item) would be the departments that require refridgerant. Produce might not be a neccisity to be refridgerated as it can be stored at room temperature but only refridgerated to prolong shelf life so if there is a lot of demand for the produce, then it wouldn't be on the shelf for long but if there is low demand for it, then it might be better to not even have it as a selection so it might be a better idea to give the refridgerator space to the other aisles.

In [10]:
# Sanity check to see if aisles such as frozen produce are part of the frozen department or produce department
frozen_aisles = df[df['aisle'].str.contains('frozen', case=False)]['aisle'].unique()
print(frozen_aisles)

['frozen produce' 'frozen meals' 'frozen meat seafood'
 'frozen appetizers sides' 'frozen breakfast' 'frozen breads doughs'
 'frozen vegan vegetarian' 'frozen pizza' 'frozen dessert' 'frozen juice']


In [11]:
frozen_department = df[df['aisle'].str.contains('frozen', case=False)]['department'].unique()
print(frozen_department)

['frozen']


It is safe to assume that 100 products for the frozen products will be from the frozen department and maybe `missing` and `other` department.

In [86]:
# storing all frozen products into a variable
freezer_df = df[df['department']=='frozen']
freezer_df

order_id  product_id                                   product_name  \
68             7       46802                               Pineapple Chunks   
100           11       30162         Teriyaki & Pineapple Chicken Meatballs   
114           12       38050  All Natural Boneless Skinless Chicken Breasts   
118           12       29471                        Combination Pizza Rolls   
135           14         162                 Organic Mini Homestyle Waffles   
...          ...         ...                                            ...   
987176     99993       43749                 Organic Ice Cream Vanilla Bean   
987180     99993        5537   Dairy Free Coconut Milk Frozen Dessert Minis   
987184     99993       26128                           Organic Mango Chunks   
987210     99996        9434                            Bag of Large Lemons   
987215     99996       32691                              Vanilla Ice Cream   

        aisle_id                    aisle  department_id department  
68           116           frozen produce              1     frozen  
100           38             frozen meals              1     frozen  
114           34      frozen meat seafood              1     frozen  
118          129  frozen appetizers sides              1     frozen  
135           52         frozen breakfast              1     frozen  
...          ...                      ...            ...        ...  
987176        37            ice cream ice              1     frozen  
987180        37            ice cream ice              1     frozen  
987184       116           frozen produce              1     frozen  
987210        34      frozen meat seafood              1     frozen  
987215        37            ice cream ice              1     frozen  

[68202 rows x 7 columns]

In [87]:
# storing all fridge products into a variable
fridge_df = df[df['department'].isin(['dairy eggs', 'meat seafood', 'deli'])]
fridge_df

order_id  product_id  \
0              1       49302   
1              1       11109   
7              1       22035   
8              2       33120   
17             3       33754   
...          ...         ...   
987241     99999       10034   
987242     99999       44085   
987243     99999        9558   
987253    100000       30489   
987254    100000       30169   

                                             product_name  aisle_id  \
0                                        Bulgarian Yogurt       120   
1           Organic 4% Milk Fat Whole Milk Cottage Cheese       108   
7                             Organic Whole String Cheese        21   
8                                      Organic Egg Whites        86   
17      Total 2% with Strawberry Lowfat Greek Strained...       120   
...                                                   ...       ...   
987241                Reduced Fat Mozarella String Cheese        21   
987242                   Vanilla Light & Fit Greek Yogurt       120   
987243               Non-Fat Vanilla Blended Greek Yogurt       120   
987253                                    Original Hummus        67   
987254            Total 2% All Natural Plain Greek Yogurt       120   

                       aisle  department_id  department  
0                     yogurt             16  dairy eggs  
1       other creams cheeses             16  dairy eggs  
7            packaged cheese             16  dairy eggs  
8                       eggs             16  dairy eggs  
17                    yogurt             16  dairy eggs  
...                      ...            ...         ...  
987241       packaged cheese             16  dairy eggs  
987242                yogurt             16  dairy eggs  
987243                yogurt             16  dairy eggs  
987253  fresh dips tapenades             20        deli  
987254                yogurt             16  dairy eggs  

[218064 rows x 7 columns]

In [89]:
# removing the freezer and fridge products from the main df as this will be for the 1000 in aisle product selection
remove_index = freezer_df.index.union(fridge_df.index)
filtered_df = df.drop(remove_index)
filtered_df

order_id  product_id                            product_name  \
2              1       10246                   Organic Celery Hearts   
3              1       49683                          Cucumber Kirby   
4              1       43633    Lightly Smoked Sardines in Olive Oil   
5              1       13176                  Bag of Organic Bananas   
6              1       47209                    Organic Hass Avocado   
...          ...         ...                                     ...   
987252    100000       19508                          Corn Tortillas   
987255    100000       38734                    Wheat Sandwich Thins   
987256    100000       36759  Unscented Long Lasting Stick Deodorant   
987257    100000       37107                            Ground Cumin   
987258    100000       31506                  Extra Virgin Olive Oil   

        aisle_id                 aisle  department_id     department  
2             83      fresh vegetables              4        produce  
3             83      fresh vegetables              4        produce  
4             95   canned meat seafood             15   canned goods  
5             24          fresh fruits              4        produce  
6             24          fresh fruits              4        produce  
...          ...                   ...            ...            ...  
987252       128  tortillas flat bread              3         bakery  
987255       128  tortillas flat bread              3         bakery  
987256        80            deodorants             11  personal care  
987257       104     spices seasonings             13         pantry  
987258        19         oils vinegars             13         pantry  

[700993 rows x 7 columns]

In [90]:
# Sanity Check to see if the rows were removed correctly
freezer_df.shape[0] + fridge_df.shape[0] + filtered_df.shape[0] == df.shape[0]

True

In [91]:
# Check to see if the right departments are removed
filtered_df['department'].unique()

array(['produce', 'canned goods', 'pantry', 'bakery', 'personal care',
       'snacks', 'breakfast', 'beverages', 'household', 'international',
       'dry goods pasta', 'babies', 'pets', 'alcohol', 'bulk', 'missing',
       'other'], dtype=object)

It did remove the aisle 

## 2. Product Selection

In [12]:
# import the library needed to split
from sklearn.model_selection import train_test_split

# splitting the data 
train_data, test_data = train_test_split(df, test_size=0.2, random_state=1)

In [13]:
# Sanity check
train_data

order_id  product_id  \
893885     90781       33731   
51142       5194        7051   
738442     74808       47626   
6338         641       47626   
922685     93610        8424   
...          ...         ...   
491263     49775       42002   
791624     80274       18628   
470924     47761       42828   
491755     49823       48946   
128037     12963       38256   

                                             product_name  aisle_id  \
893885                                    Grated Parmesan        21   
51142   Happy Tot Banana, Peach, Prune & Coconut Organ...        92   
738442                                        Large Lemon        24   
6338                                          Large Lemon        24   
922685                                     Broccoli Crown        83   
...                                                   ...       ...   
491263                       Pasture Raised Salted Butter        36   
791624             Organic Volcano Rice Ready To Eat Bowl         4   
470924                               Whipped Cream Cheese       108   
491755                                      Earl Grey Tea        94   
128037                  Maple Glazed Pecan & Sea Salt Bar         3   

                       aisle  department_id       department  
893885       packaged cheese             16       dairy eggs  
51142      baby food formula             18           babies  
738442          fresh fruits              4          produce  
6338            fresh fruits              4          produce  
922685      fresh vegetables              4          produce  
...                      ...            ...              ...  
491263                butter             16       dairy eggs  
791624         instant foods              9  dry goods pasta  
470924  other creams cheeses             16       dairy eggs  
491755                   tea              7        beverages  
128037   energy granola bars             19           snacks  

[789807 rows x 7 columns]

In [14]:
# Sanity Check
test_data

order_id  product_id  \
138856     14036       10321   
918287     93161       46584   
154109     15573        2228   
955032     96760       23695   
226812     22918        7781   
...          ...         ...   
254904     25752        8390   
120733     12254       10243   
364439     36866        5547   
642727     65142       33787   
877844     89113        9058   

                                             product_name  aisle_id  \
138856           Grands! Flaky Layers Buttermilk Biscuits       105   
918287       YoKids Blueberry & Strawberry/Vanilla Yogurt       120   
154109                        Organic Frozen Mango Chunks       116   
955032                           California Veggie Burger        42   
226812  Organic Sticks Low Moisture Part Skim Mozzarel...        21   
...                                                   ...       ...   
254904                                    Diet Ginger Ale        77   
120733                                    Parchment Paper        85   
364439                      Roasted Sesame Seaweed Snacks        66   
642727   Total 2% Lowfat Greek Strained Yogurt with Peach       120   
877844                            Gluten Free Pizza Crust       128   

                             aisle  department_id     department  
138856  doughs gelatins bake mixes             13         pantry  
918287                      yogurt             16     dairy eggs  
154109              frozen produce              1         frozen  
955032     frozen vegan vegetarian              1         frozen  
226812             packaged cheese             16     dairy eggs  
...                            ...            ...            ...  
254904                 soft drinks              7      beverages  
120733                food storage             17      household  
364439                 asian foods              6  international  
642727                      yogurt             16     dairy eggs  
877844        tortillas flat bread              3         bakery  

[197452 rows x 7 columns]

In [21]:
# Checking to see if a product appears multiple time in one order if multiple of the product is ordered 
df[df[['order_id', 'product_id']].duplicated()]

Empty DataFrame
Columns: [order_id, product_id, product_name, aisle_id, aisle, department_id, department]
Index: []

### Method 2: Product Selection Proportional to Aisle Selection

This second method is to get a variety of items from each aisle proportional to the amount of orders for that aisle.

In [24]:
# finding the total amount of products ordered
train_size = train_data.shape[0]
train_size

789807

In [47]:
# finding the amount of items ordered for each aisle
aisle_df=train_data.groupby('aisle').size().reset_index(name='count').sort_values(by='count', ascending=False)
aisle_df

aisle  count
50                 fresh fruits  88835
53             fresh vegetables  83118
98   packaged vegetables fruits  43197
133                      yogurt  35165
93              packaged cheese  23726
..                          ...    ...
76             kitchen supplies    213
2              baby accessories    186
3           baby bath body care    184
8                        beauty    134
58                 frozen juice    116

[134 rows x 2 columns]

In [48]:
aisle_df['percentage']=aisle_df['count']/train_size*100
aisle_df

aisle  count  percentage
50                 fresh fruits  88835   11.247685
53             fresh vegetables  83118   10.523837
98   packaged vegetables fruits  43197    5.469311
133                      yogurt  35165    4.452354
93              packaged cheese  23726    3.004025
..                          ...    ...         ...
76             kitchen supplies    213    0.026969
2              baby accessories    186    0.023550
3           baby bath body care    184    0.023297
8                        beauty    134    0.016966
58                 frozen juice    116    0.014687

[134 rows x 3 columns]

In [53]:
aisle_df

aisle  count  percentage
50                 fresh fruits  88835   11.247685
53             fresh vegetables  83118   10.523837
98   packaged vegetables fruits  43197    5.469311
133                      yogurt  35165    4.452354
93              packaged cheese  23726    3.004025
..                          ...    ...         ...
76             kitchen supplies    213    0.026969
2              baby accessories    186    0.023550
3           baby bath body care    184    0.023297
8                        beauty    134    0.016966
58                 frozen juice    116    0.014687

[134 rows x 3 columns]

In [68]:
train_aisle_mapping = train_data[['aisle', 'department']]
train_aisle_mapping = train_aisle_mapping.drop_duplicates()
train_aisle_mapping

aisle     department
893885               packaged cheese     dairy eggs
51142              baby food formula         babies
738442                  fresh fruits        produce
922685              fresh vegetables        produce
745108                          milk     dairy eggs
...                              ...            ...
395417            ice cream toppings         snacks
792140           baby bath body care         babies
740798              kitchen supplies      household
205099  bulk grains rice dried goods           bulk
905656                     first aid  personal care

[134 rows x 2 columns]

order_id  product_id                               product_name  \
86782       8859       30675                   Edamame Soybeans in Pods   
584410     59226       44753      New York Super Fudge Chunk® Ice Cream   
447570     45412        1609  Egg, Uncured Bacon & Cheese Scramble Cups   
622031     63014       13378                        Stir Fry Vegetables   
244156     24680        5449                           Margherita Pizza   
...          ...         ...                                        ...   
128896     13044        1986    Steamers Healthy Vision Vegetable Blend   
723457     73315       38274               Ice Cream Sandwiches Vanilla   
699870     70944       38959               Smoothies, Strawberries Wild   
21758       2231        2164            Waffles, Sweet Belgian, Vanilla   
436973     44314        9595             Organic Homestyle Mini Waffles   

        aisle_id             aisle  department_id department  
86782        116    frozen produce              1     frozen  
584410        37     ice cream ice              1     frozen  
447570        38      frozen meals              1     frozen  
622031       116    frozen produce              1     frozen  
244156        79      frozen pizza              1     frozen  
...          ...               ...            ...        ...  
128896       116    frozen produce              1     frozen  
723457        37     ice cream ice              1     frozen  
699870       113      frozen juice              1     frozen  
21758         52  frozen breakfast              1     frozen  
436973        52  frozen breakfast              1     frozen  

[54567 rows x 7 columns]

In [74]:
# removing frozen products from dataframe
mask = (train_data)
filtered_df = train_data.drop(column=['department'])

order_id  product_id                               product_name  \
86782       8859       30675                   Edamame Soybeans in Pods   
584410     59226       44753      New York Super Fudge Chunk® Ice Cream   
447570     45412        1609  Egg, Uncured Bacon & Cheese Scramble Cups   
622031     63014       13378                        Stir Fry Vegetables   
244156     24680        5449                           Margherita Pizza   
...          ...         ...                                        ...   
128896     13044        1986    Steamers Healthy Vision Vegetable Blend   
723457     73315       38274               Ice Cream Sandwiches Vanilla   
699870     70944       38959               Smoothies, Strawberries Wild   
21758       2231        2164            Waffles, Sweet Belgian, Vanilla   
436973     44314        9595             Organic Homestyle Mini Waffles   

        aisle_id             aisle  department_id department  
86782        116    frozen produce              1     frozen  
584410        37     ice cream ice              1     frozen  
447570        38      frozen meals              1     frozen  
622031       116    frozen produce              1     frozen  
244156        79      frozen pizza              1     frozen  
...          ...               ...            ...        ...  
128896       116    frozen produce              1     frozen  
723457        37     ice cream ice              1     frozen  
699870       113      frozen juice              1     frozen  
21758         52  frozen breakfast              1     frozen  
436973        52  frozen breakfast              1     frozen  

[54567 rows x 7 columns]

893885    False
51142     False
738442    False
6338      False
922685    False
          ...  
491263    False
791624    False
470924    False
491755    False
128037    False
Name: department, Length: 789807, dtype: bool

In [69]:
# Merge the 'aisle_count' DataFrame with the 'department' column from 'train_data' based on the 'aisle' column
merged_df = aisle_df.merge(train_aisle_mapping, on='aisle', how='left')
merged_df

aisle  count  percentage     department
0                  fresh fruits  88835   11.247685        produce
1              fresh vegetables  83118   10.523837        produce
2    packaged vegetables fruits  43197    5.469311        produce
3                        yogurt  35165    4.452354     dairy eggs
4               packaged cheese  23726    3.004025     dairy eggs
..                          ...    ...         ...            ...
129            kitchen supplies    213    0.026969      household
130            baby accessories    186    0.023550         babies
131         baby bath body care    184    0.023297         babies
132                      beauty    134    0.016966  personal care
133                frozen juice    116    0.014687         frozen

[134 rows x 4 columns]

As can be seen aisle 

In [17]:
df['count']

KeyError: 'count'